src/gpu/generic/sycl/ref_deconvolution.cpp

```diff
@@ -0,0 +1,132 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "gpu/generic/sycl/ref_deconvolution.hpp"
+#include "gpu/generic/sycl/convolution_kernels.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace generic {
+namespace sycl {
+
+status_t ref_deconvolution_bwd_weights_t::pd_t::init_conf() {
+    conf_ = sycl_convolution_conf_t();
+
+    conf_.diff_dst_md = xpu::sycl::md_t(src_md());
+    if (with_bias()) {
+        conf_.diff_bias_md = xpu::sycl::md_t(diff_weights_md(1));
+    }
+    conf_.data_md = xpu::sycl::md_t(diff_dst_md());
+    conf_.ndims = ndims();
+
+    memory_desc_t diff_weights_md_copy = *diff_weights_md(0);
+
+    //IC and OC are the other way around compared to convolution
+    bool no_groups = diff_weights_md(0)->ndims == diff_dst_md()->ndims;
+    auto &strides = diff_weights_md_copy.format_desc.blocking.strides;
+
+    auto recalc_strides_swap_dims = [&](int dim0, int dim1) {
+        int bigger_stride_idx = strides[dim0] > strides[dim1] ? dim0 : dim1;
+        int smaller_stride_idx = strides[dim0] > strides[dim1] ? dim1 : dim0;
+        for (int i = 0; i < diff_weights_md(0)->ndims; i++) {
+            if (strides[smaller_stride_idx] < strides[i]
+                    && strides[i] < strides[bigger_stride_idx]) {
+                strides[i] /= diff_weights_md_copy.dims[bigger_stride_idx];
+                strides[i] *= diff_weights_md_copy.dims[smaller_stride_idx];
+            }
+        }
+    };
+
+    if (no_groups) {
+        std::swap(strides[0], strides[1]);
+        recalc_strides_swap_dims(0, 1);
+        std::swap(diff_weights_md_copy.dims[0], diff_weights_md_copy.dims[1]);
+    } else {
+        std::swap(diff_weights_md_copy.dims[1], diff_weights_md_copy.dims[2]);
+        recalc_strides_swap_dims(1, 2);
+        std::swap(strides[1], strides[2]);
+    }
+
+    conf_.diff_weights_md = xpu::sycl::md_t(&diff_weights_md_copy);
+
+    // XXX: should probably be tuned.
+    conf_.block_size = 16;
+    conf_.wg_size = 32;
+
+    conf_.wk_size = memory_desc_wrapper(diff_weights_md()).nelems();
+
+    conf_.do_scale_dst
+            = !attr()->scales_.get(DNNL_ARG_SRC_0).has_default_values();
+    conf_.do_scale_weights
+            = !attr()->scales_.get(DNNL_ARG_WEIGHTS).has_default_values();
+    conf_.do_scale_data
+            = !attr()->scales_.get(DNNL_ARG_DST).has_default_values();
+    conf_.single_weight_scale
+            = attr()->scales_.get(DNNL_ARG_WEIGHTS).mask_ == 0;
+
+    conf_.use_dst_zeropoints
+            = !attr()->zero_points_.has_default_values(DNNL_ARG_SRC_0);
+    conf_.use_data_zeropoints
+            = !attr()->zero_points_.has_default_values(DNNL_ARG_DST);
+    conf_.single_dst_zeropoint = attr()->zero_points_.common(DNNL_ARG_SRC_0);
+    conf_.single_data_zeropoint = attr()->zero_points_.common(DNNL_ARG_DST);
```

Comment:
I guess this could be simplified, as bwd by weights typically does not support quantization (so no scales/zero-point).

Reply:
removed